tests/deposit_test.cairo

```diff
@@ -0,0 +1,353 @@
+    let (token_bridge, _, messaging_mock) = deploy_token_bridge_with_messaging();
+    let usdc_address = deploy_erc20("usdc", "usdc");
+    let usdc = IERC20Dispatcher { contract_address: usdc_address };
+
+    snf::start_cheat_caller_address(usdc_address, OWNER());
+    usdc.transfer(snf::test_address(), 100);
+    snf::stop_cheat_caller_address(usdc_address);
+
+    enroll_token_and_settle(token_bridge, messaging_mock, usdc_address);
+
+    usdc.approve(token_bridge.contract_address, 100);
+    token_bridge.deposit(usdc_address, 100, snf::test_address());
+
+    snf::start_cheat_caller_address(
+        token_bridge.contract_address, contract_address_const::<'user2'>()
+    );
+    token_bridge.deposit_cancel_request(usdc_address, 100, snf::test_address(), 2);
+}
+
+
+#[test]
+fn deposit_with_message_cancel_request_ok() {
+    let (token_bridge, mut spy, messaging_mock) = deploy_token_bridge_with_messaging();
+    let usdc_address = deploy_erc20("usdc", "usdc");
+    let usdc = IERC20Dispatcher { contract_address: usdc_address };
+
+    snf::start_cheat_caller_address(usdc_address, OWNER());
+    usdc.transfer(snf::test_address(), 100);
+    snf::stop_cheat_caller_address(usdc_address);
+
+    enroll_token_and_settle(token_bridge, messaging_mock, usdc_address);
+
+    let mut calldata = ArrayTrait::new();
+    'param1'.serialize(ref calldata);
+    'param2'.serialize(ref calldata);
+
+    usdc.approve(token_bridge.contract_address, 100);
+    token_bridge.deposit_with_message(usdc_address, 100, snf::test_address(), calldata.span());
+
+    token_bridge
+        .deposit_with_message_cancel_request(
+            usdc_address, 100, snf::test_address(), calldata.span(), 2
+        );
+
+    let expected_deposit_cancel = TokenBridge::DepositWithMessageCancelRequest {
+        sender: snf::test_address(),
+        token: usdc_address,
+        amount: 100,
+        appchain_recipient: snf::test_address(),
+        message: calldata.span(),
+        nonce: 2
+    };
+
+    spy
+        .assert_emitted(
+            @array![
+                (
+                    token_bridge.contract_address,
+                    Event::DepositWithMessageCancelRequest(expected_deposit_cancel)
+                )
+            ]
+        );
+}
+
+
+#[test]
+#[should_panic(expected: ('NO_MESSAGE_TO_CANCEL',))]
+fn deposit_with_message_cancel_request_no_deposit() {
+    let (token_bridge, _, messaging_mock) = deploy_token_bridge_with_messaging();
+    let usdc_address = deploy_erc20("usdc", "usdc");
+
+    enroll_token_and_settle(token_bridge, messaging_mock, usdc_address);
+
+    let mut calldata = ArrayTrait::new();
+    'param1'.serialize(ref calldata);
+    'param2'.serialize(ref calldata);
+
+    token_bridge
+        .deposit_with_message_cancel_request(
+            usdc_address, 100, snf::test_address(), calldata.span(), 2
+        );
+}
+
+#[test]
+#[should_panic(expected: ('NO_MESSAGE_TO_CANCEL',))]
+fn deposit_with_message_cancel_request_different_user() {
```

Comment:
should we do the same as `accept_deposit` here? we move the `start_message_cancellation` call to a new function called `cancel_message`, call that function from `deposit_with_message_cancel_request` and `deposit_cancel_request` and write similar tests

Reply:
Can you mention this one in the issue as well?

Reply:
Added this to issue #8